### PR TITLE
Add proposed LSP 3.16 changes

### PR DESCRIPTION
**Pull request by KamasamaK**
Some of the larger proposed LSP 3.16 changes such as call hierarchy and semantic token support have already been added or are in progress. I have bundled most of the remaining changes here. They include:

* Add client capability for resolving text edits on completion items.
* Add client capability for using client default behavior on renames. (The new `prepareRename` result still needs to be implemented)
* Add support for insert and replace ranges on `CompletionItem`.
* Add support for diagnostic code descriptions.
* Add support for document symbol provider label.
* Add support for tags on `SymbolInformation` and `DocumentSymbol` and deprecated the standalone `deprecated` properties.
* Add support for call hierarchy item `data` property.
* Add support for code action `data` property.
* Add support for code action `disabled` property.
* Add support for code action resolve request.
* Add support for diagnostic `data` property.
* Add support for signature information `activeParameter` property.

There are also some minor changes to comments or format.

### CODE REVIEW

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 1]
```diff
@@ -537,6 +612,27 @@ class SymbolKindCapabilities {
 	}
 }
 
+/**
+ * Specific capabilities for the `SymbolTag`.
+ */
```
Comment (Dufgui) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:617]:
```suggestion
 * Since 3.16.0
 */
```

Comment (Dufgui) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:617]: I see you already put it when it's use, not sure, it's usefull their.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 2]
```diff
@@ -1008,7 +1199,7 @@ class TypeHierarchyCapabilities extends DynamicRegistrationCapabilities {
 }
 
 /**
- * Capabilities specific to the {@code textDocument/callHierarchy}.
+ * Capabilities specific to the {@code textDocument/prepareCallHierarchy}.
  */
```
Comment (Dufgui) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:1203]:
```suggestion
 * Since 3.16.0
 */
```

**Comment by Dufgui** [comment]
Huge and wonderfull work. Hyper clean ;o)

I think we can add "since" on:
- CallHierarchyPrepareParams  
- CallHierarchyIncomingCallsParams
- CallHierarchyOutgoingCallsParams 
- CallHierarchyIncomingCall 
- CallHierarchyOutgoingCall 
- CallHierarchyItem

**Review by Dufgui — CHANGES_REQUESTED** [review]
Huge and wonderfull work. Hyper clean ;o)

I think we can add "since" on:
- CallHierarchyPrepareParams  
- CallHierarchyIncomingCallsParams
- CallHierarchyOutgoingCallsParams 
- CallHierarchyIncomingCall 
- CallHierarchyOutgoingCall 
- CallHierarchyItem

**Comment by Dufgui** [comment]
maybe we can add the Since in `TextDocumentService ` for callHierarchy calls ?

**Review by spoenemann — APPROVED** [review]
Great, thank you!

**Pull request merged**